apps/backoffice/src/components/services/adapters.ts

```diff
@@ -228,11 +227,11 @@ const getCtaUrlFromCtaString = (value: string) => {
  * or on future uses of such fields which may be undefined.
  * @param s service
  * @returns service without epmpty or undefined properties */
-const clearUndefinedEmptyProperties = (s: any): any => {
+const clearUndefinedNullEmptyProperties = (s: any): any => {
```

Comment:
this is the only method in this repo that requires lodash. lodash provides the same features as fp-ts in a more imperative fashion. I strongly suggest to remove lodash in order to reduce the size of the bundle and rewrite this method. You can use plain javascript:

```ts
/** Remove temporary assistanceChannels[] metadata used for react-hook-form dynamic array fields. */
const removeAssistanceChannelsArray = (s: any) => {
  const { assistanceChannels, ...metadata } = s.metadata;
  return { ...s, metadata };
};

type DeepPartial<T> = {
  [P in keyof T]?: DeepPartial<T[P]>;
};

function clearUndefinedNullEmptyProperties<T extends object>(
  obj: T,
  filterOutValues = new Set([null, undefined, ""])
): DeepPartial<T> {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    if (filterOutValues.has(value)) {
      return acc;
    }

    if (typeof value === "object" && !Array.isArray(value)) {
      return { ...acc, [key]: clearUndefinedNullEmptyProperties(value) };
    }

    return { ...acc, [key]: value };
  }, {});
}

```

Reply:
Hi @gunzip, I actually use lodash also in another component _(for object deep comparison)_.
In general I consider `lodash` a very valid _(and widely used)_ library. 
Furthermore, IMO, it is convenient and powerful as it expresses concepts in a minimal way.
However, I accept your objection and I have opened an internal technical debt [task](https://pagopa.atlassian.net/browse/IOPAE-902) to evaluate the problem globally on the repo.

Reply:
Can you share a link with the code where lodash is used?

Reply:
https://github.com/pagopa/io-services-cms/blob/b1e6dec1059b7b75b02c96c667e37ade2ac2b91e/apps/backoffice/src/components/create-update-process/create-update-process.tsx#L103

Reply:
You can use something alike

```ts
export const isDeepEqual = (a: any, b: any): boolean => {
  if (a === b) {
    return true;
  }

  if (typeof a !== typeof b) {
    return false;
  }

  if (typeof a !== "object" || a === null || b === null) {
    return false;
  }

  if (Object.keys(a).length !== Object.keys(b).length) {
    return false;
  }

  if (Array.isArray(a) !== Array.isArray(b)) {
    return false;
  }

  if (Array.isArray(a)) {
    return a.every((item, index) => isDeepEqual(item, b[index]));
  }

  return Object.keys(a).every(key => isDeepEqual(a[key], b[key]));
};

```

or

```ts
export const isDeepEqual = (a: any, b: any): boolean => {
  return JSON.stringify(a) === JSON.stringify(b);
};

```

Reply:
Sorry but it's not the same.
json.stringify assumes that the strings are ordered in the same way _(too ugly to look at! )_ while in the other case it doesn't seem acceptable to me to write about 20 lines of codes _(which also have to be tested)_ to do a job that I can achieve with one line...it seems more like a trial of intentions than a real need to eliminate a library.
Anyway we will evaluate this possibility in the technical debt task I reported above, but I would like to let you know that I do not agree with the approach

Reply:
In which case 20 lines of code are worse than serving an entire library just to use 2 methods? Furthermore, consider in your evalutation that 80% of lodash features are covered by fp-ts which is a library you're already using.

Reply:
I will evaluate it in internal technical debt [task](https://pagopa.atlassian.net/browse/IOPAE-902).
Thank you so much for the valuable advice and food for thought, as always 🙏😉